Allow stricter typed dispatch with redux-thunk

Unfortunately I failed using ng2-redux in combination with ThunkAction from redux-thunk using typescript.

The problem is, that dispatch expects an Action but ThunkAction does not extend Action.

I agree with you. But wouldn't that soften the interface contract to much? 

redux-thunk redeclares the Dispatch<S> function of redux (see: https://github.com/gaearon/redux-thunk/blob/master/index.d.ts#L8) to accept ThunkAction. 

I tried to do so with NgRedux

```
declare module 'ng2-redux' {
    export interface NgRedux<S> {
        dispatch<R, E, A extends Action>(action: ThunkAction<R, S, E> | A): any;
    }
}
```

 but failed with  

> error TS2484: Export declaration conflicts with exported declaration of 'NgRedux'

Tried using any and it works, but then you loose the ThunkAction type, trying to use ThunkAction type it complains since you can't override the Action type:
```
error TS2344: Type '(dispatch: Dispatch<any>, getState: () => any, extraArgument: any) => any' does not satisfy the constraint 'Action'.
  Property 'type' is missing in type '(dispatch: Dispatch<any>, getState: () => any, extraArgument: any) => any'
```
Maybe making the dispatch generic argument not extends from Action?

--- a/src/components/ng-redux.ts
+++ b/src/components/ng-redux.ts
@@ -8,7 +8,8 @@ import {
   Unsubscribe,
   createStore,
   applyMiddleware,
-  compose
+  compose,
+  Dispatch,
 } from 'redux';
 
 import { NgZone } from '@angular/core';
@@ -166,7 +167,7 @@ export class NgRedux<RootState> {
   /**
    * Dispatch an action to Redux
    */
-  dispatch = <A extends Action>(action: A): any => {
+  dispatch: Dispatch<RootState> = action => {
     if (!this._store) {
       throw new Error('Dispatch failed: did you forget to configure your store? ' +
         'https://github.com/angular-redux/@angular-redux/core/blob/master/' +

